lib/src/shared/codec_registry.dart

```diff
@@ -0,0 +1,37 @@
+import 'codec.dart';
+
+/// Encloses classes related to the compression and decompression of messages.
+class CodecRegistry {
+  CodecRegistry({List<Codec> codecs = const [const Identity()]})
+      : assert(codecs != null) {
+    codecs.forEach((codec) => register(codec));
+  }
+
+  factory CodecRegistry.empty() {
+    return CodecRegistry(codecs: []);
+  }
+
+  /// Key refers to the `messageEncoding` string from the [Codec].
+  final Map<String, Codec> _codecs = {};
+  final List<String> _allSupportedCodecs = [];
+
+  Codec lookupCodec(String codecName) {
+    return _codecs[codecName];
+  }
+
+  String encodings() {
+    return _allSupportedCodecs.join(',');
+  }
+
+  /// Registers a [Codec] for both decompression and message encoding
+  /// negotiation.
+  void register(Codec codec) {
```

Comment:
Is adding codecs dynamically a required API for your needs? If not - I'd consider dropping it. Requiring to construct registry with all supported codecs from the very beginning seems reasonable.

Reply:
Removed for now.. I added it originally to keep in sync with the java api but it doesn't seem reasonable.